tests/tom-swifty-test/PosixUtilsTests.cs

```diff
@@ -40,11 +40,12 @@ public void DotDotWorks1 ()
 		public void RelativePathsInSymlinks1 ()
 		{
 			using (var dir = new DisposableTempDirectory ()) {
-				var expectedPath = Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (dir.DirectoryPath)));
+				var expectedPath = "/private" + Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (dir.DirectoryPath)));
```

Comment:
That is strange, look at the following example: 
<img width="659" alt="Screenshot 2024-02-12 at 16 21 25" src="https://github.com/xamarin/binding-tools-for-swift/assets/2190086/1927ab16-6ea1-4743-a3ca-2882c8b12c9f">

This is running the mono RELP but I'd be surprised that dotnet does to do the right things. Can you show us what was the code you tested?

Reply:
I tried:
```
Path.Combine ("/private", Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (dir.DirectoryPath))));
```
and
```
Path.Combine ("/private/", Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (dir.DirectoryPath))));
```

Reply:
If the second path is an absolute path, then the second path is returned as-is.

```cs
Path.Combine ("/a", "/b"); // prints "/b"
Path.Combine ("/a", "b"); // prints "/a/b"
```

Also note that the `/private` prefix is because `Path.GetTempPath ()` returns something like `/var/folders/..`, where `/var ` is a symlink to `/private/var`:

```shel
$ ls -lad /var
lrwxr-xr-x@ 1 root  wheel  11 Jan 12 08:19 /var -> private/var
```

so maybe a better fix would be to do something like this?

```cs
				var realDirectoryPath = PosixHelpers.RealPath (dir.DirectoryPath);
				var expectedPath = Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (realDirectoryPath)));
```